sycl/source/detail/device_global_map.hpp

```diff
@@ -27,7 +27,10 @@ class DeviceGlobalMap {
   DeviceGlobalMap(bool OwnerControlledCleanup)
       : MOwnerControlledCleanup{OwnerControlledCleanup} {}
 
-  ~DeviceGlobalMap() {
+  DeviceGlobalMap(const DeviceGlobalMap &) = default;
+  DeviceGlobalMap &operator=(const DeviceGlobalMap &) = default;
+
+  ~DeviceGlobalMap() noexcept(false) {
```

Comment:
Coverity is complaining about C++ destructor implicitly marked as `noexcept` while the destructor body can throw an exception, in `DeviceGlobalIt.second->cleanup();`.
I've marked the destructor as `noexcept(false)` to indicate that destructor can throw

Reply:
I'm not sure this is a good solution. I tried it once and it just lead to more Coverity hits.

If this class is destroyed in any OTHER classes destructor, then the problem recurs. Do you mark that destructor as `noexcept(false)`  too?   

I guess many of our impl classes now have try/catch blocks and they stream out errors. But should that occur it'll likely be difficult to trace the problem back to here.

Anyway, I haven't thought about this deeply, and perhaps in this situation you have. Just wanted to check.